File: OMS-Service/migrations/status-change-history.js
'use strict';
module.exports = {
  up: async (queryInterface, Sequelize) => {
    await queryInterface.createTable(
      'status_change_history',
      {
        id: {
          allowNull: false,
          autoIncrement: true,
          primaryKey: true,
          type: Sequelize.INTEGER
        },
        order_id: {
          type: Sequelize.STRING,
          allowNull: false
        },
        split_order_id: {
          type: Sequelize.STRING,
          allowNull: true
        },
        split_order_increment_id: {
          type: Sequelize.STRING,
          allowNull: true
        },
        processing_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        pending_payment_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        cancel_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        packed_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        shipped_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        rto_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        delivered_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        picked_up_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        refunded_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        received_warehouse_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        rma_cancel_date: {
          type: 'TIMESTAMP',
          allowNull: true,
          defaultValue: '0000-00-00 00:00:00'
        },
        created_at: {
          allowNull: false,
          type: Sequelize.DATE,
          defaultValue: Sequelize.fn('NOW')
        },
        updated_at: {
          allowNull: false,
          type: Sequelize.DATE,
          defaultValue: Sequelize.fn('NOW')
        }
      },
      {
        underscored: true
      }
    );
  },
  down: async queryInterface => {
    await queryInterface.dropTable('status_change_history');
  }
};

File: java-api/order-service/src/main/java/org/styli/services/order/model/sales/StatusChangeHistory.java
package org.styli.services.order.model.sales;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Timestamp;

/**
 * Salesrule Coupon
 */
@Data
@Entity
@Table(name = "status_change_history")
public class StatusChangeHistory implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", insertable = false, nullable = false)
    private Integer id;

    @Column(name = "order_id", nullable = false)
    private String orderId;
    @Column(name = "order_increment_id", nullable = false)
    private String orderIncrementId;
  
    @Column(name = "processing_date")
    private Timestamp processingDate;
    
    @Column(name = "pending_payment_date")
    private Timestamp pendingPaymentDate;
    
    @Column(name = "cancel_date")
    private Timestamp cancelDate;
   
    @Column(name = "packed_date")
    private Timestamp packedDate;
    
    @Column(name = "shipped_date")
    private Timestamp shippedDate;
       
    @Column(name = "rto_date")
    private Timestamp rtoDate;
    
    @Column(name = "delivered_date")
    private Timestamp deliveredDate;
    
    @Column(name = "rma_created_at")
    private Timestamp rmaCreatedDate;
      
    @Column(name = "picked_up_date")
    private Timestamp pickedUpDate;
    
    @Column(name = "refunded_date")
    private Timestamp refundedDate;
       
    @Column(name = "received_warehouse_date")
    private Timestamp receivedWarehouseDate;
    
    @Column(name = "rma_cancel_date")
    private Timestamp rmaCancelDate;
    
    @Column(name = "created_at")
    private Timestamp createAt;
    
    @Column(name = "updated_at")
    private Timestamp updatedAt;

    @Column(name = "split_order_id", nullable = true)
    private Integer splitOrderId;

    @Column(name = "split_order_increment_id", nullable = true)
    private String splitOrderIncrementId;
}